src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -2096,5 +2116,34 @@ private Guid IdToGuidConverter(string id)
 
             return deterministicGuid;
         }
+
+        internal bool GetMatchingWorkspaceData(string uniqueId, out Dictionary<string, string> data)
+        {
+            data = new Dictionary<string, string>();
+            if (!ExtensionData.Any())
+                return false;
+
+            var extensionData = ExtensionData.Where(x => x.ExtensionGuid == uniqueId).FirstOrDefault();
+            if (extensionData is null)
+                return false;
+
+            data = extensionData.Data;
+            return true;
+        }
+
+        internal void UpdateExtensionData(string uniqueId, Dictionary<string, string> data)
+        {
+            var extensionData = ExtensionData.Where(x => x.ExtensionGuid == uniqueId).FirstOrDefault();
+            if (extensionData is null)
+                return;
+            extensionData.Data = data;
+        }
+
+        internal void CreateNewExtensionData(string uniqueId, string name, Dictionary<string, string> data)
+        {
+            // TODO: Figure out how to add extension version when creating new ExtensionData 
+            var extensionData = new ExtensionData(uniqueId, name, "NotSureHowWeShouldDoThis...", data);
```

Comment:
I forgot what we decided here, I think we said grab the version number from the extension assembly?